### ISSUE TITLE
react/display-name triggers on certain function components  

### ISSUE THREAD

**Issue by elijahsmith** [yannickcr/eslint-plugin-react#277]
for example, this simple class is flagged (adapted from React docs):

```
import React from 'react';

var Aquarium = (props) => {
    var fish = props.species.toLowercase();
    return <div>{fish}</div>;
};

export default Aquarium;
```

from exploration, it seems that if you don't simply return JSX from a function component -- that is, if your component *doesn't* look like:

```
var Aquarium = (props) => (<div>{props.fish}</div>);
```

then the rule is triggered.

I've even had it triggered by having functions defined inside the JSX, as so:

```
var Aquarium = (props) => (
  <div>
    {props.fishes.map(fish => {
      return fish;
    })}
  </div>
```

...somewhat contrived but I hope the concept is conveyed...

**Comment by yannickcr** [comment]
Working on it

**Comment by yannickcr** [comment]
Should be fixed by 2749a0b15bdc7747fe30cdebfde7f54cd41ba75e in `v3.7.0`. Let me know if it's not the case.

Also, if you do not want to set a `displayName` to your stateless components, I suggest you to set the [acceptTranspilerName option](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/display-name.md#accepttranspilername) to `true`.

**Comment by elijahsmith** [comment]
awesome, thank you @yannickcr
